### PR TITLE
Qt: Fix games crashing on start

### PR DESCRIPTION
**Pull request by spycrab**
Fixes a regression caused by the button mapping windows

### CODE REVIEW

**Comment by JosJuice** [comment]
Is this a fix for https://bugs.dolphin-emu.org/issues/10300 ?

**Comment by spycrab** [comment]
Yes. But I thought that Qt Bugs didn't belong in the bug tracker?

**Comment by JosJuice** [comment]
If it's in the https://github.com/dolphin-emu/dolphin repo, it belongs on the bug tracker.

**Comment by spycrab** [comment]
@JosJuice Well...the UI itself says so:
Screenshot

But I guess that's up for discussion (although I'd prefer to discuss this in IRC instead of adding more nonsensical comments to this PR)

File: Source/Core/DolphinQt2/MainWindow.cpp [hunk 1]
```diff
@@ -38,13 +45,27 @@ MainWindow::MainWindow() : QMainWindow(nullptr)
...
 {
   m_render_widget->deleteLater();
 }
 
+void MainWindow::InitControllers()
+{
+  if (g_controller_interface.IsInit())
+    return;
+
+  g_controller_interface.Initialize(reinterpret_cast<void*>(reinterpret_cast<void*>(winId())));
```
Comment (lioncash) [inline Source/Core/DolphinQt2/MainWindow.cpp]: aren't one of the casts here redundant?

Comment (spycrab) [inline Source/Core/DolphinQt2/MainWindow.cpp]: Looks like it.

Comment (spycrab) [inline Source/Core/DolphinQt2/MainWindow.cpp]: Fixed it.

**Pull request merged**